rapthor/lib/field.py

```diff
@@ -200,6 +165,91 @@ def scan_observations(self, data_fraction=1.0):
+                        self.observations.append(Observation(obs.ms_filename,
+                                                             starttime=midpoint-chunktime/2,
+                                                             endtime=midpoint+chunktime/2))
+                    else:
+                        self.observations.append(obs)
+                else:
+                    steptime = mintime * (tottime / mintime - nchunks) / nchunks + mintime
+                    starttimes = np.arange(obs.starttime, obs.endtime, steptime)
+                    endtimes = np.arange(obs.starttime+mintime, obs.endtime+mintime, steptime)
+                    for starttime, endtime in zip(starttimes, endtimes):
+                        if endtime > obs.endtime:
+                            starttime = obs.endtime - mintime
+                            endtime = obs.endtime
+                        self.observations.append(Observation(obs.ms_filename, starttime=starttime,
+                                                             endtime=endtime))
+        minnobs = self.parset['cluster_specific']['max_nodes']
+        if len(self.imaging_sectors)*len(self.observations) < minnobs:
+            # To ensure all the nodes are used efficiently, try to divide up the observations
+            # to reach at least minnobs in total. For simplicity, we try to divide each existing
+            # observation into minnobs number of new observations (since there is no
+            # drawback to having more than minnobs observations in total)
+            mintime = self.parset['calibration_specific']['slow_timestep_sec']
+            prev_observations = self.observations[:]
+            self.observations = []
+            for obs in prev_observations:
+                tottime = obs.endtime - obs.starttime
+                nchunks = min(minnobs, max(1, int(tottime / mintime)))
+                if nchunks > 1:
+                    steptime = tottime / nchunks
+                    steptime -= steptime % mintime
+                    starttimes = np.arange(obs.starttime, obs.endtime, steptime)
+                    endtimes = [st+steptime for st in starttimes]
+                    for nc, (starttime, endtime) in enumerate(zip(starttimes, endtimes)):
+                        if nc == len(endtimes)-1:
+                            endtime = obs.endtime
+                        self.observations.append(Observation(obs.ms_filename, starttime=starttime,
+                                                             endtime=endtime))
+                else:
+                    self.observations.append(obs)
+
+        # Update the copies stored in the imaging sectors
+        for sector in self.imaging_sectors:
+            sector.observations = []
+            for obs in self.observations:
+                obs.log = None  # deepcopy cannot copy the log object
+                cobs = copy.deepcopy(obs)
+                obs.log = logging.getLogger('rapthor:{}'.format(obs.name))
```

Comment:
This assumes the "old" logger object was 'rapthor:<obs.name>'. Why not save the old logger in say `old_logger` and reassign that value to `obs.log` here?

Reply:
Good point -- I made a copy() method in the Observation class that should ensure there's no confusion with the logger name